aten/src/THNN/generic/SmoothL1Criterion.c

```diff
@@ -7,15 +7,17 @@ void THNN_(SmoothL1Criterion_updateOutput)(
           THTensor *input,
           THTensor *target,
           THTensor *output,
+          accreal beta_,
           int64_t reduction)
 {
   THNN_CHECK_SHAPE(input, target);
+  scalar_t beta = TH_CONVERT_ACCREAL_TO_REAL(beta_);
 
   if (reduction == Reduction::None) {
     THTensor_(resizeAs)(output, input);
     TH_TENSOR_APPLY3(scalar_t, input, scalar_t, target, scalar_t, output,
       scalar_t z = fabs(*input_data - *target_data);
-      *output_data = z < 1 ? 0.5 * z * z : z - 0.5;
+      *output_data = z < beta ? 0.5 * z * z : z - 0.5;
```

Comment:
I believe this is not enough. This makes the function discontiguous for beta different than 1.
I believe the standard way of doing this would be something like
```c
z < beta ? 0.5 * z * z / beta : z - 0.5 * beta
```

See https://github.com/facebookresearch/maskrcnn-benchmark/blob/f40f29931c7d68da696168d8fdf7332281de28a9/maskrcnn_benchmark/layers/smooth_l1_loss.py#L13 and https://en.wikipedia.org/wiki/Huber_loss for some more information

Reply:
I made some updates, now I got tests failing for all the no_reduce cases. The reason is because it fails the double gradient checks. I made some changes to the double_backwards implementation could you please take a look?

https://github.com/pytorch/pytorch/blob/87ce7c2f0ea8be5957f482ffba7f481a78b1c01c/torch/nn/_functions/thnn/auto_double_backwards.py#L206-L221